markdown/org/grading.md

```diff
@@ -7,4 +7,233 @@ hidden: true
+kompilieren und/oder ausführen lassen. Stellen Sie vor Abgabe sicher, dass Ihr Code keine fest
+codierten Dateien/Verzeichnisse enthält. Denken Sie daran, dass bei UNIX ebenso wie bei Java
+zwischen Groß- und Kleinschreibung unterschieden wird, dass Dateinamen und Klassennamen korrekt
+sein müssen, dass die angegebenen Methodensignaturen korrekt sein müssen usw. ...
+
+Die Codequalität zählt. Selbst wenn Ihre Lösung eine perfekte Funktionalität aufweist, gibt es
+Abzug für schlechte oder schlampige Programmierung. Dies betrifft u.a. fehlende/mangelhafte
+Dokumentation, Verletzung von Namenskonventionen, uneinheitliche Formatierung,
+schlechte Namen für Klassen/Methoden/Variablen/..., duplizierter Code, ...
+
+Konkret können Sie für die Nichteinhaltung der formalen Kriterien folgende Abzüge pro Abgabe
+"erreichen":
+*   Keine/nicht ausreichende Tests: -2P
+*   Kein/nicht ausreichendes Logging: -1P
+*   Kein/nicht ausreichendes Javadoc: -1P
+*   Code-Style nicht/nur ungenügend beachtet: -1P
+
+### Lerntagebuch
+
+Sie führen zu jeder Abgabe von Aufgaben aus Pool A/B/C oder P ein Lerntagebuch. Wenn dies komplett
+nicht vorhanden ist oder nur ungenügend ausgearbeitet ist, bekommen Sie auf die gesamte Abgabe der
+betreffenden Aufgaben keine Punkte (also nur noch für die Pool-K-Aufgaben).
+
+Bei der Abgabe der Pool-P-Aufgaben bekommen Sie zusätzlich für Ihr Lerntagebuch Punkte.
+
+<!-- TODO: Inhalte und Struktur Lerntagebuch -->
+
+### Besonderheit Pool P: "Projekt"
+
+Pool P steht für "Projekt". Sie können diese Aufgabe maximal zweimal bearbeiten und jeweils maximal
+20 Punkte bekommen. Die Bearbeitungszeit ist jeweils für zwei Wochen angelegt.
+
+Spätestens eine Woche vor dem Projekt stellen Sie im Praktikum Ihr Konzept dem Lehrbeauftragten vor
+und diskutieren mit ihm die Machbarkeit und die Verteilung der Punkte.
+
+In der ersten Woche der Bearbeitung stellen Sie im Praktikum Ihr Feinkonzept und eine erste grobe
+Basis-Implementierung vor. Dafür können Sie maximal 10 Punkte bekommen.
+
+In der zweiten Woche stellen Sie dann Ihre finale Implementierung, die Dokumentation und das Lerntagebuch
+vor und bekommen für die sinnvolle Umsetzung die zu Beginn vereinbarten Teilpunkte. Es gelten die selben
+Kriterien für die Abgabe wie für die Pool A/B/C-Aufgaben, d.h. Sie können entsprechende Abzüge für das
+Nichteinhalten formaler Kriterien bekommen.
+
+<!-- TODO: Kriterien für schriftliche Ausarbeitung -->
+
+Die Projektaufgaben können Sie frühestens in der Woche nach dem E-Assessment bearbeiten, d.h. Sie müssten
+dann entsprechend Ihr Konzept in der Woche vor dem E-Assessment oder in der Woche mit dem E-Assessment
+vorstellen.
+
+### Praktikum: Zeitslots
+
+Wir werden mit jedem Team einen individuellen Abgabeslot in Ihrer Praktikumszeit vereinbaren, damit Sie
+im Praktikum zielgerichtet zu Ihrer Abgabe kommen können und so Wartezeiten vermeiden. Dazu gibt es
+vorbereitete Etherpads im ILIAS, wo Sie sich mit Ihrem Team für einen Abgabeslot eintragen.
+
```

Comment:
Hinweis darauf, dass es sich um eine Online Abgabe per Zoom handelt?

Reply:
Das müssten wir tatsächlich nochmal disktuieren.

Ich bin jetzt erstmal davon ausgegangen, dass die Praktika komplett online sind. Aber wenn ich die Mail der Chefin korrekt lese, dann können wir auch Hybrid anbieten. Ich werde das mal entsprechend ergänzen.

Reply:
Die Frage ist, ob Jonas darüber so richtig glücklich ist. Er muss ja trotzdem in die FH und dann die Abnahme von dort dann machen ...

Reply:
done.